das_client/lib/sfera/src/mapper/track_equipment_mapper.dart

```diff
@@ -0,0 +1,143 @@
+import 'package:das_client/model/journey/track_equipment.dart';
+import 'package:das_client/sfera/src/mapper/sfera_model_mapper.dart';
+import 'package:das_client/sfera/src/model/enums/start_end_qualifier.dart';
+import 'package:das_client/sfera/src/model/enums/track_equipment_type.dart';
+import 'package:das_client/sfera/src/model/network_specific_area.dart';
+import 'package:das_client/sfera/src/model/segment_profile.dart';
+import 'package:das_client/sfera/src/model/segment_profile_list.dart';
+import 'package:fimber/fimber.dart';
+
+class TrackEquipmentMapper {
+  TrackEquipmentMapper._();
+
+  static List<NonStandardTrackEquipmentSegment> parseNonStandardTrackEquipmentSegment(
+      List<SegmentProfileList> segmentProfilesLists, List<SegmentProfile> segmentProfiles) {
+    final trackEquipments = _parseTrackEquipments(segmentProfilesLists, segmentProfiles);
+    trackEquipments.sort((a, b) => a.compareTo(b));
+
+    final openStartSegments = <SferaTrackEquipmentType, _NonStandardTrackEquipment?>{};
+
+    final List<NonStandardTrackEquipmentSegment> segments = [];
+    for (final trackEquipment in trackEquipments) {
+      if (trackEquipment.startLocation != null && trackEquipment.endLocation != null) {
+        segments.add(_createSegmentFromStartsEnds(trackEquipment));
+      } else if (trackEquipment.startLocation != null) {
+        if (openStartSegments.containsKey(trackEquipment.type)) {
+          continue;
+        }
+        openStartSegments[trackEquipment.type] = trackEquipment;
+      } else if (trackEquipment.endLocation != null) {
+        final startOfSegment = openStartSegments[trackEquipment.type];
+        if (startOfSegment != null) {
```

Comment:
als else eine Warnung loggen?

Reply:
Wir müssen dann mal noch den Fall prüfen, wenn ein "Starts" im letzten oder ein "Ends im ersten Segment kommt. Bin mir nicht sicher, ob das fachlich möglich ist.

Reply:
There we go 😭 

<img width="880" alt="image" src="https://github.com/user-attachments/assets/02544907-f999-4bc2-a583-58e2d031e103">